superset-frontend/src/dashboard/actions/hydrate.js

```diff
@@ -302,6 +302,13 @@ export const hydrateDashboard = (dashboardData, chartData, datasourcesData) => (
     filterSetsConfig: metadata?.filter_sets_configuration || [],
   });
 
+  if (!metadata) {
+    metadata = {};
+  }
+
+  metadata.show_native_filters =
```

Comment:
@simcha90 @amitmiran137 @villebro should this use the `DASHBOARD_NATIVE_FILTERS` feature flag for defining the fallback rather than defaulting to `true`? We've seen instances at Airbnb where dashboard native filters are showing up in the dashboard metadata even though the feature is disabled.